sprockets_statsd/statsd.py

```diff
@@ -0,0 +1,435 @@
+    already formatted payloads.
+
+    .. attribute:: host
+       :type: str
+
+       IP address or DNS name for the statsd server to send metrics to
+
+    .. attribute:: port
+       :type: int
+
+       TCP port number that the statsd server is listening on
+
+    .. attribute:: should_terminate
+       :type: bool
+
+       Flag that controls whether the background task is active or
+       not.  This flag is set to :data:`False` when the task is started.
+       Setting it to :data:`True` will cause the task to shutdown in
+       an orderly fashion.
+
+    .. attribute:: queue
+       :type: asyncio.Queue
+
+       Formatted metric payloads to send to the statsd server.  Enqueue
+       payloads to send them to the server.
+
+    .. attribute:: running
+       :type: asyncio.Event
+
+       Is the background task currently running?  This is the event that
+       :meth:`.run` sets when it starts and it remains set until the task
+       exits.
+
+    .. attribute:: stopped
+       :type: asyncio.Event
+
+       Is the background task currently stopped?  This is the event that
+       :meth:`.run` sets when it exits and that :meth:`.stop` blocks on
+       until the task stops.
+
+    """
+
+    protocol: typing.Union[StatsdProtocol, None]
+    _create_transport: typing.Callable[[], typing.Coroutine[
+        typing.Any, typing.Any, typing.Tuple[asyncio.BaseTransport,
+                                             StatsdProtocol]]]
+
+    def __init__(self,
+                 *,
+                 host,
+                 port: int = 8125,
+                 reconnect_sleep: float = 1.0,
+                 ip_protocol: int = socket.IPPROTO_TCP,
+                 wait_timeout: float = 0.1):
+        super().__init__()
+        if not host:
+            raise RuntimeError('host must be set')
+        try:
+            port = int(port)
+            if not port or port < 1:
+                raise RuntimeError(
+                    f'port must be a positive integer: {port!r}')
+        except (TypeError, ValueError):
+            raise RuntimeError(f'port must be a positive integer: {port!r}')
+
+        transport_creators = {
+            socket.IPPROTO_TCP: self._create_tcp_transport,
+            socket.IPPROTO_UDP: self._create_udp_transport,
+        }
+        try:
+            self._create_transport = transport_creators[ip_protocol]
+        except KeyError:
+            raise RuntimeError(f'ip_protocol {ip_protocol} is not supported')
+
+        self.host = host
+        self.port = port
+        self._ip_protocol = ip_protocol
+        self._reconnect_sleep = reconnect_sleep
+        self._wait_timeout = wait_timeout
+
+        self.running = asyncio.Event()
+        self.stopped = asyncio.Event()
+        self.stopped.set()
+        self.logger = logging.getLogger(__package__).getChild('Processor')
+        self.should_terminate = False
+        self.protocol = None
+        self.queue = asyncio.Queue()
```

Comment:
Worth thinking about the healthy expectation here for metric collection in the event that it can't make a connection to the host, should metrics continue to be recorded until the application runs into issue?

Reply:
I ended up putting a parameterized max size on the queue (0f0bae1) with a default of 1000 elements.  We can change the default in the future if it is too conservative.